File: Math/523-contactBinary.py
from math import log2
class Solution:
    def concatenatedBinary1(self, n: int) -> int:
        arr=[bin(i)[2:] for i in range(1,n+1)]
        return int("".join(arr),2)%(pow(10,9)+7)
    def concatenatedBinary2(self, n: int) -> int:
        ans=0
        b=0
        M=10**9+7
        for i in range(1,n+1):
            if i&(i-1)==0:    
                b+=1
            ans<<=b
            ans+=i
            ans%=M

        
        return ans
    
    def isPowerofTwo1(self,n):
        x=int(log2(n))
        return 2**x==n
    
    def isPowerOfTwo2(self,n):
        return log2(n).is_integer()

File: Math/523-contactBinary.cpp

#include<iostream>
#include<cmath>
using std::cout;


int concatenateBin(int n)
{
    int ans=0;
    int b=0;
    int MOD=std::pow(10,9)+7;
    for(int i=1;i<n+1;++i)
    {
        if((i&(i-1))==0)
            b+=1;
        ans<<=b;
        ans+=i;
        ans%=MOD;
    }
    return ans;
}

int main()
{
    cout<<concatenateBin(3);
}